addons/mail/demo/discuss/public_channel_demo.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
+
+        <record model="mail.guest" id="mail.public_guest">
+            <field name="name">Guest Alex</field>
+        </record>
+
+        <!-- Discuss Public Channel -->
+        <record model="discuss.channel" id="mail.channel_public">
+            <field name="name">Public Community</field>
+            <field name="description">A space for engaging discussions among employees, partners, and the public.</field>
+            <field name="group_public_id" eval="False"/>
+        </record>
+
+        <!-- Messages of Public Channel -->
+        <record model="mail.message" id="mail.channel_public_message_0">
+            <field name="model">discuss.channel</field>
+            <field name="res_id" ref="mail.channel_public"/>
+            <field name="message_type">comment</field>
+            <field name="subtype_id" ref="mail.mt_comment"/>
+            <field name="author_id" ref="base.partner_admin"/>
+            <field name="body">We're excited to announce our product launch this September! 🎉
+            We'll be hosting a live video conference on this channel to showcase all the details.
+            If you're interested, feel free to share the link with your customers so they can tune
+            in and watch it live!</field>
+        </record>
+        <record model="mail.message" id="mail.channel_public_message_1">
+            <field name="model">discuss.channel</field>
+            <field name="res_id" ref="mail.channel_public"/>
+            <field name="message_type">comment</field>
+            <field name="subtype_id" ref="mail.mt_comment"/>
+            <field name="author_guest_id" ref="mail.public_guest"/>
+            <field name="parent_id" ref="mail.channel_public_message_0"/>
+            <field name="body">That sounds amazing! Can't wait to see what's new.
+            Will there be a QA session during the conference?</field>
+        </record>
+        <record model="mail.message" id="mail.channel_public_message_2">
+            <field name="model">discuss.channel</field>
+            <field name="res_id" ref="mail.channel_public"/>
+            <field name="message_type">comment</field>
+            <field name="subtype_id" ref="mail.mt_comment"/>
+            <field name="author_id" ref="base.partner_demo"/>
+            <field name="parent_id" ref="mail.channel_public_message_1"/>
+            <field name="body">Thanks, Alex! 😊 Yes, there will be a live QA session at
+            the end of the conference where attendees can ask questions and get insights
+            directly from our team. Stay tuned for more details!</field>
+        </record>
+        <record model="discuss.channel.member" id="mail.admin_member_channel_public">
```

Comment:
Maybe also some reactions on the messages while we are at it? I don't think we already have some of them in demo data.